Assets/Scripts/Weapon/HarpoonGun/HarpoonSpear.cs

```diff
@@ -0,0 +1,127 @@
+using Unity.VisualScripting;
+using UnityEngine;
+using Enemies;
+using Omnia.Utils;
+/*
+    The projectile for HarpoonGun
+    This class should only be interacted upon by its prefabs and HarpoonGun
+*/
+public class HarpoonSpear : MonoBehaviour {
+    public LayerMask enemyLayer;
+    public LayerMask playerLayer;
+    public LayerMask groundLayer;
+
+    public Collider2D Collider2D;
+    public Rigidbody2D Rigidbody2D;
+
+    private bool dropped;
+    private HarpoonGun gun;
+
+    // Tracking enemy
+    private Enemy taggedEnemy;
+    public Enemy TaggedEnemy {
+        get { return taggedEnemy; }
```

Comment:
Could collapse this into one property getter with a autogenerated backing field

public Enemy TaggedEnemy {get; private set;}